src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -44,7 +44,23 @@ internal static string EmitQueryParameterPreparation(this EndpointParameter endp
                         {
                             wasParamCheckFailure = true;
                         }
-                        var {{endpointParameter.HandlerArgument}} = {{endpointParameter.Name}}_raw.ToString();
+                        var {{endpointParameter.Name}}_temp = {{endpointParameter.Name}}_raw.ToString();
+""");
+        }
+
+        if (endpointParameter.IsParsable)
```

Comment:
Can the emit for the parsability logic be moved to a separate method so that it can be invoked from other codepaths that might have a `TryParse` (`FromHeader`, `FromRoute`, etc)

Reply:
I did this although I don't think it buys us much. Once we do array handling we'll probably end up breaking it up again.